orbit/cmd/orbit/orbit.go

```diff
@@ -930,11 +933,26 @@ func main() {
 			orbitClient.RegisterConfigReceiver(extRunner)
 		}
 
+		// Run a early check of fleetd configuration to check if orbit needs to
+		// restart before proceeding to start the sub-systems.
+		//
+		// E.g. the administrator has updated the following agent options for this device:
+		//	- `update_channels`
+		//	- `extensions` were removed/unset
+		//	- `command_line_flags` (osquery startup flags)
 		if err := orbitClient.RunConfigReceivers(); err != nil {
 			log.Error().Msgf("failed initial config fetch: %s", err)
+		} else {
+			if orbitClient.RestartTriggered() {
+				log.Info().Msg("exiting after early config fetch")
+				return nil
+			}
```

Comment:
This is the bug fix. We were missing an early check for restart before starting sub-systems.